src/opm/input/eclipse/Schedule/Network/NetworkKeywordHandlers.cpp

```diff
@@ -170,6 +172,29 @@ void handleNODEPROP(HandlerContext& handlerContext)
         }
 
         if (as_choke) {
+            const auto& group = handlerContext.state().groups.get(name);
+            if (group.wellgroup()) {
+                //Wells belong to a group with autochoke enabled are to be run on a common THP and should not have guide rates
+                for (const std::string& wellName : group.wells()) {
+                    auto well = handlerContext.state().wells.get(wellName);
+
+                    //Let the wells be operating on a THP Constraint
+                    auto properties = std::make_shared<Well::WellProductionProperties>(well.getProductionProperties());
+                    //The wells are not to be under GRUP control using guide rates but under THP control
+                    properties->dropProductionControl(Well::ProducerCMode::GRUP);
```

Comment:
We do not need to drop the `GRUP` control.  The following code will make the group control unavailable. 
```c++
 well.updateAvailableForGroupControl(false);
```

The good thing about not dropping it is that with new `NODEPROP` keyword, when the node is not autochoke anymore, we do not have to add `GRUP` back.

Reply:
Good point.  The drop of `GRUP` control is dropped.